index.js

```diff
@@ -36,88 +36,187 @@ module.exports = function(options) {
      * @api public
      */
 
-    this.onLoad(extRegex, function(file, next) {
-      utils.matter.parse(file, next);
-    });
+    this.onLoad(extRegex, utils.mu.series([
+      isHandled,
+      isNull,
+      isBinary,
+      utils.matter.parse,
+      escape,
+      stripPrefixes,
+      utils.file.renameFile(this)
+    ]));
+
+    this.onLoad(jsonRegex, parseJson);
 
     /**
-     * Uses C-style macros to escape templates with `{%%= foo %}` or
-     * `<%%= foo %>` syntax, so they will not be evaluated by a template
-     * engine when `.render` is called.
+     * Registers a `.preWrite` middleware for unescaping escaped delimiters.
      *
-     * @name escape templates
+     * @name unescape templates
      * @api public
      */
 
-    this.onLoad(escapeRegex, function(file, next) {
-      var str = file.contents.toString();
-      str = str.replace(/([{<])(%%=?)/g, '__ESC_$1DELIM$2__');
-      file.contents = new Buffer(str);
-      next();
-    });
+    this.postRender(escapeRegex, utils.mu.series([
+      utils.matter.stringify,
+      unescape
+    ]));
 
-    this.preWrite(escapeRegex, unescape);
+    this.preWrite(jsonRegex, updateJson);
+    return plugin;
+  };
+};
 
-    /**
-     * Adds a `json` property to the `file` object when the file extension
-     * matches `options.jsonRegex`. This allows JSON files to be updated
-     * by other middleware or pipeline plugins without having to parse and
-     * stringify with each modification.
-     *
-     * @name JSON on-load
-     * @api public
-     */
+function isHandled(file, next) {
+  file.isHandled = function(name, stage) {
+    this.handled = this.handled || {};
+    if (this.handled.hasOwnProperty(name)) {
+      if (typeof stage !== 'string') return true;
+      return this.handled[name].indexOf(stage) !== -1;
```

Comment:
I think this line should be:

```js
if (this.handled[name].indexOf(stage) !== -1) return true;
```

This will add `stage` to the `this.handled[name]` array below and return false.